src/content/blog/node-test-migration.mdx

```diff
@@ -0,0 +1,133 @@
+
+Once the PR was ready, we noticed that _Node.js test runner_ **was significantly slower** than Mocha. We investigated, and we discovered that Node.js spawns a new process for each test file to assure that each testing suite is run in **isolation**. Running a testing suite in isolation is, generally, a good practice because it assures that tests run in an unpolluted environment.
+
+However, our testing suites were already isolated. In fact, we were able to run our testing suites with Mocha using the main thread without running into any of the typical issues: side effects, polluted environments, etc. Unfortunately, Node.js didn't provide an option to _run all tests in the same thread_. So, we had to come up with a solution to the slow test runs. (Aren't we engineers, after all? We solve problems!)
+
+Using our internal `astro-scripts test` command, we were able to workaround this by creating a _temporary file that **imports** all the testing suites_, and we let Node.js test that single file. This way, only one process is spawned for the file and we reach the same level of performance as if we were using the main process.
+
+However, this **came with its own downside**: if there's a test failure or a timeout, we weren't able to tell which test was the cause. This was the main quirk we found, and while not every team may have made this choice, we accepted this trade-off in order to give us the benefits mentioned earlier. After all, we had previously accepted Mocha's quirks!
+
+### Node.js `assert` and `chai`
+
+During the migration, we had to remove the `chai` library for `node:assert/strict`. This task uncovered that with `chai`, you can execute the same check in different ways. For example, you can run an equality check at least in **four** different ways:
+
+```js
+import { expect } from "chai";
+
+expect("foo").to.eq("foo")
+expect("foo").to.be.eq("foo")
+expect("foo").to.equal("foo")
+expect("foo").to.be.equal("foo")
+```
+
+On the one hand, it's good to have this kind of flexibility. But on the other, the code for the tests becomes inconsistent. With the Node.js assertion module, there is only one way to perform this check:
+
+```js
+import { assert } from "node:assert/strict";
+
+assert.equal("foo", "foo")
+```
+
+The Node.js assertion module **provides almost all of the functionalities** we required, so the migration from `chai` wasn't as painful as we thought it might be. Our use of `chai` was very minimal. However, **we do miss** the `.includes` shortcut of `chai`:
+
+```js
+import { expect } from "chai";
+
+expect("It's a fine day").includes("fine")
+```
+
+The Node.js assertion module doesn't provide such a utility, so instead we ended up using the equality assertion with the `String#includes` function:
+
+```js
+import assert from "node:assert/strict";
+
+assert.equal("It's a fine day".includes("fine"), true)
+```
+
+### Here comes the dragons
```

Comment:
[According to George R.R. Martin](https://georgerrmartin.com/notablog/2022/05/05/here-come-the-dragons/), there's no `s` in `come` here!  
```suggestion
### Here come the dragons
```